docs/spec/slashing/04_begin_block.md

```diff
@@ -108,6 +110,17 @@ for val in block.Validators:
     signInfo.MissedBlocksCounter--
   // else previous == val not in block.AbsentValidators, no change
 
+	// Emit warning events if Validator misses block
+	if val in block.AbsentValidators {
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeSlash,
+				sdk.NewAttribute(types.AttributeKeyAddress, consAddr.String()),
+				sdk.NewAttribute(types.AttributeKeyMissedBlocks, fmt.Sprintf("%d", signInfo.MissedBlocksCounter)),
```

Comment:
although this is correct, it's very golang-specific. I'd maybe get rid of the `fmt.Sprintf` and packages (`types.` and `sdk.`)